.bazelrc

```diff
@@ -5,6 +5,14 @@
 ## https://bazel.build/docs/best-practices#bazelrc-file
 ###############################################################################
 
+# These are workspaces that are tested using rules_bazel_integration_test. The
+# files need to be found by the parent workspace for packaging, but we do not
+# want the child workspace packages to be evaluated by the parent workspace.
+# To update these lines, execute 
+# `bazel run @rules_bazel_integration_test//tools:update_deleted_packages`
+build --deleted_packages=bzlmod/hello_world
```

Comment:
Should we just pull all your changes here into examples/bzlmod/hello_world?

Reply:
This PR allows someone to run `bazel test //bzlmod:hello_world` from the root module. Underneath the covers, it will launch another Bazel process and execute tests in the `bzlmod/hello_world` workspace. I use this pattern in [rules_swift_package_manager](https://github.com/cgrindel/rules_swift_package_manager/blob/main/bzlmod/BUILD.bazel) and [bazel-starlib](https://github.com/cgrindel/bazel-starlib/blob/main/examples/BUILD.bazel). In essence, one can run all tests for a  child workspace from the parent workspace. 

For this to work, the parent workspace needs to see the files for the child workspace, but not process them as Bazel packages. That is what `--deleted_packages` does. The other way to prevent the parent workspace from processing the child workspaces is to add them to `.bazelignore`. However, this makes those directories completely inaccessible to the parent.

I noticed that in `rules_rust` the child workspaces are ignored by the parent workspace by adding them to `.bazelignore`. The child workspaces are exercised by changing into the directories and running `bazel test //...`.  

So, the question for the maintainers is whether they are interested in being able to test the child workspaces from the parent workspace. If so, then we need the `--deleted_packages` and the `filegroup(name="all_files")`. Otherwise, we can use the pattern that is currently being used for other child workspaces in this repo.

@matts1 @illicitonion What do you prefer?

Reply:
One benefit of collecting the runtime files in a filegroup and testing them using `rules_bazel_integration_test` is that you have the list of files that should be included in the release archive. [In `rules_swift_package_manager`](https://github.com/cgrindel/rules_swift_package_manager/blob/main/release/BUILD.bazel#L54-L61), I create the release archive using the same files that are used for the `rules_bazel_integration_test` tests.

Reply:
> So, the question for the maintainers is whether they are interested in being able to test the child workspaces from the parent workspace.

I'm not especially interested in this - I find it much easier to debug something when it's not a generated workspace, and having to manage the filegroups seems like a steep price to pay for me. That being said, I'll defer to to others, since I'm not really representative of most developers since I'm very familiar with the bzlmod examples and the workflows for dealing with changes that affect bzlmod.

> One benefit of collecting the runtime files in a filegroup and testing them using rules_bazel_integration_test is that you have the list of files that should be included in the release archive.

I'm personally quite fine with just tarring up everything except `examples`, unless you have some justification otherwise.

Reply:
I'd lean towards another child workspace, because of the simplicity it provides - not needing the filegroup, being able to trivially just `cd` in to debug, etc. I don't have a strong objection to the `--deleted_packages` dance - open to the discussion, but would prefer another child workspace absent that discussion :)

Reply:
No problem. I will modify the PR and ping folks when it is ready.

Reply:
> I'm personally quite fine with just tarring up everything except examples, unless you have some justification otherwise.

For repos that declare dev dependencies in their `MODULE.bazel`, you would not want to include any files that reference the dev dependencies. It can cause issues with queries executed by a client workspace, because the dev dependencies are not loaded. I don't see any dev dependencies in this repo. So, it should not matter.